app/routes/api.generate-image-caption.ts

```diff
@@ -0,0 +1,34 @@
+import type { ActionFunctionArgs } from '@remix-run/node';
```

Comment:
Probably, I missed the idea. What is the purpose of calling this route starting with api? Why should it be in the routes folder?